### ISSUE TITLE
Low starting accuracy when training with Vehicle loss functions

### ISSUE THREAD

**Issue by wenkokke** [vehicle-lang/vehicle#431]
@ndslusarz  Could you provide some details? What test reproduces the behaviour, what test establishes the baseline.

**Comment by ndslusarz** [comment]
This problem occurs when running `vehicle-python/tests/test_mnist` in the branch `loss_experiments`. Training without calling custom loss (merely with cross-entropy loss alone - as is in the current file the relevant lines, that is 58 and latter half of 60, calling loss translation is commented out) starts with first epoch getting test accuracy: 0.0533

Now as a baseline the same neural network, with data imported the same way - but using standard keras code - gets accuracy 0.93 on first epoch so a difference large enough that there has to be some mistake in the code. The code for this one can be found in `standard-mnist-training` in the same folder for comparison but it is just a very standard keras model.

**Comment by wenkokke** [comment]
I’m not entirely sure what two setups you mean. Would it be possible for you to commit them as two files? Are these two steps meant to be using *exactly the same* loss functions and training code, or should the variant using the custom loss be *slightly different but not this bad*?

**Comment by ndslusarz** [comment]
I've done that now. 

There is the code for running `vehicle-python/tests/test_mnist_cross_entropy.py` - this one uses the python code with a custom loss but does not actually call for any constraint (does not call vehicle itself at all). It merely uses cross entropy but through the manual training loop writen for our loss.

The file `vehicle-python/tests/standar-mnist-training.py` just has a python and keras script, using standard keras functions, for training a model with the exact same architecture and data.

I had this as a sanity check - to check that provided no fancy custom loss out training behaves similiarly to how it would normally do in a standard keras implementation so it's the comparison between those two.

**Comment by wenkokke** [comment]
We could try rewriting both programs in the style of the standard program, by passing the custom loss function in to the `model.compile` call? That’d help us rule out other causes. The loss parameter can be an instance of `tf.losses.Loss`, so we could use that for our custom function. See the following example from the [documentation](https://www.tensorflow.org/api_docs/python/tf/keras/losses/Loss):
```python
class MeanSquaredError(tf.losses.Loss):

  def call(self, y_true, y_pred):
    return tf.reduce_mean(tf.math.square(y_pred - y_true), axis=-1)
```

**Comment by Tgl70** [comment]
This is now fixed. When changing the accuracy function it was mistakenly changed to a loss, hence it was going down instead of up.
